### PR TITLE
Attempt isValidEnumValue to the semantic of TEnum::GetEnum.

### PR DESCRIPTION
**Pull request by pcanal**
And avoid nullptr dereference

### CODE REVIEW

File: roofit/roofitcore/src/RooFactoryWSTool.cxx [hunk 1]
```diff
@@ -241,13 +241,12 @@ namespace {
...
 
-    // FIXME: We don't have any guarantee that typeName is less than 256 bytes
-    // Chop type name into class name and enum name
-    char buf[256];
-    strlcpy(buf,typeName,256);
-    char* className = strtok(buf,":");
-
-    return (strcmp(TEnum::GetEnum(className)->GetName(), value) == 0);
+    auto enumType = TEnum::GetEnum(typeName);
+    if (enumType) {
```
Comment (vgvassilev) [inline roofit/roofitcore/src/RooFactoryWSTool.cxx]: That's too many braces for my eyes: what about `return (enumType) : enumType->GetConstant(value) ? false;`

Comment (pcanal) [reply roofit/roofitcore/src/RooFactoryWSTool.cxx]: sure why not ...

**Comment by pcanal** [comment]
@yamaguchi1024 On the previous attempt what test/functionality failed?

**Comment by yamaguchi1024** [comment]
tutorial-roofit-rf511_wsfactory
tutorial-roofit-rf511_wsfactory_basic
tutorial-roofit-rf513_wsfactory_tools

I think it was these three (Sorry for the late reply)

**Comment by pcanal** [comment]
@phsft-bot build

**Comment by pcanal** [comment]
roottest_root_multicore_tp_process_imt crashes within Davix.  root_multicore_processExecutorH1Test crashes without showing a stack trace (in the web output).  So I assume both are network related.  merging.

**Pull request merged**